### DISCUSSION TITLE
Accessibility metadata in epub content.opf

### DISCUSSION THREAD

**Discussion by Iacobus1983** [jgm/pandoc#9365] [Ideas]
In view of [EU Accessibility Act](https://www.w3.org/TR/epub-a11y-eaa-mapping/), I checked a pandoc-generated epub with [DAISY Ace (1.3.2)](https://daisy.github.io/ace/). I get some accessibility warnings. 

In order to make the epubs compliant with accessibility requirements, it seems that minor addition to `content.opf` would be needed:

1. `xml:lang` attribute should be added to `package` element, most likely with the same value of `dc:language` element (see https://kb.daisy.org/publishing/docs/wcag/language-of-page.html)
2. [Accessibility `meta` elements](https://kb.daisy.org/publishing/docs/metadata/schema.org/index.html): 
 - `schema:accessMode`, `schema:accessibilityFeature` and `schema:accessibilityHazard` are mandatory;
 - `schema:accessModeSufficient` and `schema:accessibilitySummary` are optional.

Full support for metadata mentioned in point 2 would possibly require a user-defined input, e.g. within the metadata yaml file (perhaps similar to https://github.com/jgm/pandoc/commit/743d1b368f8c8bbd393454915f71a02453659423 ?)

**Comment by jgm** [comment]
I've made this an issue.
#9372
Further comments can go there.

**Comment by Qchristensen** [comment]
I see the linked issue was merged in version 3.1.12.1.

I added the following metadata to my source HTML file:
```xml
<metadata>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">

<meta name="dcterms:conformsTo" content="EPUB Accessibility 1.1 - WCAG 2.1 Level AA"/>
<meta name="schema:accessMode" content="textual"/>
<meta name="schema:accessModeSufficient" content="textual"/>
<meta name="schema:accessibilityFeature" content="none"/>
<meta name="schema:accessibilityHazard" content="none"/>
<meta name="schema:accessibilitySummary" content="This publication strives to meet accepted Web Content Accessibility Guidelines (WCAG) at the AA level. Page numbers are present and navigation to pages is supported. This publication contains a comprehensive multi-level table of contents for navigation through the various chapters and sections."/>
</metadata>
```
When converting to ePub file this all used to be missed.  In 3.1.12.1 now everything is brought into the ePub except the schema:accessibilitySummary property.  Is this a lingering little bug with that issue, or is there something wrong with my metadata please?

Kind regards

Quentin

**Reply by jgm** [reply]
I don't understand. `<metadata>` isn't a valid HTML element.  If you want to use an HTML source for your epub, while adding some EPUB metadata, then this is not and never has been the way to do it.  The manual tells you how to add metadata to an EPUB.  You can use either `--epub-metadata` with XML metadata or just `--metadata-file` with markdown/YAML metadata.  More details in the manual.

**Reply by jgm** [reply]
I don't think it's possible to add the accessibility features with `--epub-metadata`; so the way to go is to use `--metada-file` with a markdown YAML metadata block as described here: https://pandoc.org/MANUAL.html#epub-metadata  You can do that even if your source file is HTML or another format.

**Reply by Qchristensen** [reply]
Ok that all sounds complex.  I just borrowed that block off I can't even remember now and didn't analyse the <metadata> tag too much :)  But it actually does seem to work - without it, I get errors on all those other tags

**Reply by Iacobus1983** [reply]
@Qchristensen To explain why it seems to work: when converting to epub, pandoc 3.1.12.1 adds by default reasonable values for `accessMode`, `accessibilityFeature`, `accessibilityHazard` and `accessModeSufficient`. So, I guess that the metadata you have in the source html are stripped by pandoc, but you still get an accessible epub with the default values (which are stated in pandoc's manual). 

For `accessibilitySummary` there's no default, since it's supposed to be a text in the language of the book, and also because it's not mandatory according to [EPUB Accessibility 1.1](https://www.w3.org/TR/epub-a11y-11/#sec-disc-package). It can be added via metadata block.

Use of `--metadata-file` is indeed quite easy. A block of accessibility metadata would be something like that (supposing you want to change the default for `accessModes` and add an `accessibilitySummary`:
```
---
accessModes: 
- "textual" 
- "visual"
accessibilitySummary: My custom text
---
```

**Reply by Qchristensen** [reply]
Thanks!  I've got it working now at least (or at least not throwing any errors in Ace).  I'd have preferred something I could add to the original source so it would ensure I wouldn't forget it next time, but I've got the instructions written down for then at least and I can just copy the metadata file around as needed :)
